### DISCUSSION TITLE
Is it possible to select patch and voices rather using a (QWERTY) keyboard than using the rotary encoder?

### DISCUSSION THREAD

**Discussion by JustGlowing** [probonopd/MiniDexed#631] [Q&A]
Hello everyone,

I'd like to try MiniDexed without using installing the display and the rotary encoder. Is it possible to control MiniDex using a keyboard connected via usb?

I know other people have asked about using MIDI commands, but I'd prefer to use a pc keyboard for the job as my midi controller has minimal controls.

**Comment by diyelectromusic** [comment]
Not exactly... but it is possible to configure MIDI buttons to use Note messages, so if you picked out certain MIDI notes from https://github.com/probonopd/MiniDexed/blob/main/src/pckeyboard.cpp you might be able to configure them for PgmUp/PgmDown (for example)...

But I'm not sure what would happen with MIDI channels and if you could still use the keyboard for playing notes too..

Alternatively if you fancied building your own version of the code you might be able to add in additional keymappings in the same pckeyboard.cpp file to generate additional MIDI events...?

Control would always be pretty minimal though with no display as we don't support the HDMI display for the UI, just for logging.

I have no idea if any of what I've just said will work mind - I'm just thinking out loud! :)

Kevin

**Reply by diyelectromusic** [reply]
Alternatively, if you can run it on a Pi with USB gadget support you could plug it directly into a PC and use a MIDI control app/program on the PC to control it?  Details of how to do that are on the wiki.

Kevin

**Reply by JustGlowing** [reply]
Thanks, this makes sense. Maybe I'll try to hack the code at some point.

**Reply by probonopd** [reply]
Out of curiosity:
What would be your use case? I would assume almost all users to use a MIDI controller or sequencer or computer to play notes anyway?
It's always amazing to learn in which creative, unforseen ways people use MiniDexed :-)

**Reply by Banana71** [reply]
A PC keyboard would be an additional option to use the miniDexed cheaply and conveniently. It would be possible to switch between performances using the arrow keys, for example, even without an encoder and display.
Even in combination with a display and encoder, comfort would be increased with a PC keyboard. My first thought was a numeric keypad to select the patch or performance number and to use the arrow keys to browse through the menus. The rotary encoder could be used for quick fine-tuning of the parameters.

As JustGlowing understands, he wants to select sounds with the PC keyboard and play the notes with his simple MIDI controller.

![Zehnertastatur](https://github.com/probonopd/MiniDexed/assets/104296205/6c262f25-46ff-4bf0-9ad1-16db0fa2a15a)

**Comment by JustGlowing** [comment]
My midi controller only has piano keys and it's quite clunky to send midi
cc with it. I was hoping to install minidexed on the pi and control it with
a small PC keyboard while playing using the controller.

On second thoughts, I could use my phone to send midi cc signals but I'm
trying to avoid having screens around.

**Comment by JustGlowing** [comment]
If have to be frank, what's putting me off from installing the display and
the encoder is the fact that it would take me quite a bit of time and
minidex might eventually not be for me.

I guess enabling controls from a keyboard would allow me to try the setup
without having to fiddle with electronics from the beginning.

**Reply by craigyjp** [reply]
My concern is controlling with a keyboard is ok, but you still need a display to navigate properly.

**Reply by diyelectromusic** [reply]
I think in that case, give it a go with a MIDI controller or even some kind of MIDI control app on a PC just to see if you are interested in it.

I think you need to decide what is the path of least resistance to just having a go with it, and then you can decide how best to move forward if it is something you're interested in :)

Kevin
